api/app/Console/Jobs/JobBase.php

```diff
@@ -54,34 +60,102 @@ protected function log(string $message) : void
      */
     protected function stopIfTimeout(): void
     {
-        if(time() >= $this->startTimestamp + $this->maxJobTime)
+        if($this->timeoutEnabled && (time() >= $this->startTimestamp + $this->maxJobTime))
         {
-            $this->log("Finished ".get_class($this)." job by timeout.");
+            $this->log("Finished ".get_class($this)." job by timeout");
             exit;
         }
     }
-
+    
     /**
      * Extracts TableHeader and rows from table sheet.
      *
      * @param string $tableID
      * @param string $sheetName
      * @param string $quotaUserPrefix
      * @return array [ TableHeader, values ]
+     * @throws \Exception
      */
     protected function getHeaderAndDataFromTable(string $tableID, string $sheetName, string $quotaUserPrefix) : array
     {
         $headerRange = $sheetName.'!A1:FZ1';
-        $headerResponse = $this->spreadsheetClientService->getSpreadsheetCellsRange(
-            $tableID, $headerRange, $quotaUserPrefix."GH");
-        $propertyColumns = new TableHeader($headerResponse[0]);
-
-        $range = $sheetName.'!A2:FZ5001';
-        $values = $this->spreadsheetClientService->getSpreadsheetCellsRange(
-            $tableID, $range, $quotaUserPrefix."GB");
-
+        
+        try {
+            $headerResponse = $this->spreadsheetClientService->getSpreadsheetCellsRange(
+                $tableID,
+                $headerRange
+//            $quotaUserPrefix . "GH"
+            );
+            $propertyColumns = new TableHeader($headerResponse[0]);
+        } catch (\Exception $exception) {
+            $message = "Error on '". $tableID."' while getting spreadsheet headerResponse ".PHP_EOL.
+                $exception->getMessage();
+            $this->log($message);
+//            Log::error($message);
+            $this->throwExceptionIfQuota($exception);
+            
+            $propertyColumns = null;
+        }
+        
+        sleep(1);
+    
+        try {
+            $range = $sheetName.'!A2:FZ5001';
+            $values = $this->spreadsheetClientService->getSpreadsheetCellsRange(
+                $tableID,
+                $range
+//                $quotaUserPrefix."GB"
+        );
+        } catch (\Exception $exception) {
+            $message = "Error on '". $tableID."' while getting spreadsheet values".PHP_EOL.$exception->getMessage();
+            $this->log($message);
+//            Log::error($message);
+            $this->throwExceptionIfQuota($exception);
+            
+            $values = [];
+        }
+    
         return [ $propertyColumns, $values ];
     }
+    
+    /**
+     * Extracts data from table sheet.
+     *
+     * @param string $tableID
+     * @param string $sheetName
+     * @return array [ TableHeader, values ]
+     * @throws \Exception
+     */
+    protected function getFullDataFromTable(string $tableID, string $sheetName) : array
```

Comment:
Why is it in the base class when only one inheritant uses it?

Reply:
now all uses it